code/modules/clothing/clothing.dm

```diff
@@ -410,9 +410,68 @@ BLIND     // can't see anything
 	name = "suit"
 	var/fire_resist = T0C+100
 	allowed = list(/obj/item/weapon/tank/emergency_oxygen)
-	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 0, rad = 0)
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0)
+	var/armor_adjust = null //Used to make sure adjusted jackets get their armour back when closed.
 	slot_flags = SLOT_OCLOTHING
 	var/blood_overlay_type = "suit"
+	var/suit_adjusted = 0
+	var/ignore_suitadjust = 1
+
+//Proc that opens and closes jackets.
+/obj/item/clothing/suit/proc/adjustsuit(var/mob/user)

```

Comment:
This entire proc comes across as needlessly complex when all you need to do is adjust the item and icon states plus have some flavortext.

Also, buttoning up velcro reference is hilarious.